### PR TITLE
Drop Python v3.7 support, upgrade deps, add Python v3.13 support

### PR DESCRIPTION
**Pull request by jherland**
Three themes in one PR:
- Drop support for Python v3.7
- Run `poetry update` to update our dependencies within existing constraints
  - ...and fix minor fallout from upgraded dependencies
- Add support for Python v3.13

Happy to split this up into multiple PRs if that helps the review process.

Commits:
- Update FAQ to better explain running FD outside of project env
- Drop support for Python v3.7
- Remove obsolete `@calculated_once` decorator
- `poetry update`
- Fixes after upgraded `codespell`
- Reimplement `Location._sort_key` as a `@cached_property`
- Update type annotations to appease Mypy
- Fix new Ruff check by rephrasing logic and adding comments
- Apply simple Ruff fixes after upgrade
- Add support for Python v3.13

### CODE REVIEW

File: README.md
Comment (jherland) [inline README.md:644]: Add a paragraph mentioning which versions of FawltyDeps runs on v3.7.

Comment (jherland) [reply README.md:644]: Added in bf179f3.

**Review by mknorps — APPROVED** [review]
I left some happy comments here and there.
Thank you for handling that and reducing the code complexity :rocket:

**Review by obscurerichard — APPROVED** [review]
👍🏻 :shipit:

**Comment by obscurerichard** [comment]
Just before we merge this we should drop the branch protection rule that mandates tests on Python 3.7. 

That would also be a good time to rework that into a Ruleset instead of a classic Branch Protection Rule.

Are we ready to merge this otherwise @jherland ?

**Comment by jherland** [comment]
> Just before we merge this we should drop the branch protection rule that mandates tests on Python 3.7.

Yes! Thanks for reminding me.

> That would also be a good time to rework that into a Ruleset instead of a classic Branch Protection Rule.
> 
> Are we ready to merge this otherwise @jherland ?

Yes, except I wanted to make another release with Conda support added, but before we remove support for running FD on Python 3.7 (this PR). See #462 for details.

**Comment by jherland** [comment]
> Just before we merge this we should drop the branch protection rule that mandates tests on Python 3.7.

Done

> That would also be a good time to rework that into a Ruleset instead of a classic Branch Protection Rule.

I setup a new ruleset in "evaluation" mode, and left the old branch protection rule (minus 3.7 checks) in place, for now. Let's have a closer look at it together in the status meeting?

**Comment by jherland** [comment]
Pushed a minor update/rebase, mostly to trigger the new ruleset. Will merge when green.

**Pull request merged**
